piccolo/query/mixins.py

```diff
@@ -664,24 +665,36 @@ def to_string(value) -> str:
 
     @property
     def action_string(self) -> QueryString:
+        engine = engine_finder()
         action = self.action
+
         if isinstance(action, OnConflictAction):
             if action == OnConflictAction.do_nothing:
                 return QueryString(OnConflictAction.do_nothing.value)
             elif action == OnConflictAction.do_update:
                 values = []
-                query = f"{OnConflictAction.do_update.value} SET"
+                if engine is not None:
+                    if engine.engine_type == "mysql":
+                        query = ""
+                    else:
+                        query = f"{OnConflictAction.do_update.value} SET"
```

Comment:
I get linter errors in this function, because `query` is potentially undefined, and lower down we append to `query`.

Reply:
The reason I get a linter error is in this logic if we don't enter this block:

```python
if engine is not None
```
Then `query` is undefined.

You could just put in an `assert engine` and drop the `if`.

```python
assert engine
if engine.engine_type == "mysql":
    query = ""
else:
    query = f"{OnConflictAction.do_update.value} SET"
```